testnet/stacks-node/src/nakamoto_node/miner.rs

```diff
@@ -155,6 +155,27 @@ impl BlockMinerThread {
         let mut stackerdbs = StackerDBs::connect(&self.config.get_stacker_db_file_path(), true)
             .expect("FATAL: failed to connect to stacker DB");
 
+        let sort_db = SortitionDB::open(
+            &self.config.get_burn_db_file_path(),
+            true,
+            self.burnchain.pox_constants.clone(),
+        )
+        .expect("FATAL: could not open sortition DB");
+
+        let tip = SortitionDB::get_block_snapshot_consensus(
+            sort_db.conn(),
+            &self.burn_block.consensus_hash,
+        )
+        .expect("FATAL: could not retrieve chain tip")
+        .expect("FATAL: could not retrieve chain tip");
+
+        let reward_set = sort_db
+            .get_preprocessed_reward_set_of(&tip.sortition_id)
+            .expect("FATAL: Error fetching reward set")
+            .expect("FATAL: No reward set found for miner")
+            .known_selected_anchor_block_owned()
+            .expect("FATAL: No reward set found for miner");
```

Comment:
I moved the loading of the reward set out of `gather_signatures` and into this higher-level spot so that we're not loading the reward set on every single loop. I see that the code above this is mainly using `expect`s instead of returning errors, so I followed that same pattern here, but wanted to double check that that's right

Reply:
Let's remove these `.expect`s then.  The miner is going to be retrying this code over and over, so any transient problems (like my comment below) might go away later.